chart/templates/_helpers.tpl

```diff
@@ -72,6 +72,58 @@ Create the name of the job service account to use
 {{- end }}
 {{- end }}
 
+{{- define "job-executor-service.remote-control-plane.endpoint" }}
+    {{- if ((.Values.remoteControlPlane).autoDetect).enabled }}
+        {{- $detectedNamespace := include "job-executor-service.remote-control-plane.namespace" .}}
+        {{- (printf "http://api-gateway-nginx.%s/api" $detectedNamespace) }}
+    {{- else }}
+        {{- (printf "%s://%s/api" .Values.remoteControlPlane.api.protocol .Values.remoteControlPlane.api.hostname) }}
+    {{- end }}
+{{- end }}
+
+{{- define "job-executor-service.remote-control-plane.configuration-endpoint" }}
+    {{- (printf "%s/configuration-service" (include "job-executor-service.remote-control-plane.endpoint" .)) }}
+{{- end }}
+
+
+{{/*
+Helper functions of the auto detection feature of Keptn
+*/}}
+{{- define "job-executor-service.remote-control-plane.namespace" -}}
+    {{- if .Values.remoteControlPlane.autoDetect.namespace }}
+        {{- .Values.remoteControlPlane.autoDetect.namespace }}
+    {{- else }}
+    {{- $detectedKeptnNamespaces := list }}
+
+    {{- /* Find api-gateway-nginx service, which is used as keptn api gatway */ -}}
+    {{- range $index, $srv := (lookup "v1" "Service" "" "").items }}
+        {{- if and (eq $srv.metadata.name "api-gateway-nginx") (hasPrefix "keptn-" ((get $srv.metadata.labels "app.kubernetes.io/part-of") | default "")) }}
+            {{- $detectedKeptnNamespaces = append $detectedKeptnNamespaces $srv.metadata.namespace }}
+        {{- end }}
+    {{- end }}
+
+    {{- if eq (len $detectedKeptnNamespaces) 0 }}
+        {{- fail "Unable to detect Kepn in the kubernetes cluster!" }}
+    {{- end }}
+    {{- if gt (len $detectedKeptnNamespaces) 1 }}
+        {{- fail "Detected more than one Keptn installation!" }}
+    {{- end }}
+
+    {{- index $detectedKeptnNamespaces 0 }}
+    {{- end }}
```

Comment:
I would get rid of using the `.Values.remoteControlPlane.autoDetect.namespace` and always do a lookup, since that value is supposed to look into a particular namespace but I would like to still check for the api-gateway-nginx

```suggestion
    {{- $detectedKeptnApiGateways := list }}

    {{- /* Find api-gateway-nginx service, which is used as keptn api gatway */ -}}

    {{- $services := lookup "v1" "Service" (.Values.remoteControlPlane.autoDetect.namespace | default "") "" }}

    {{- range $index, $srv := $services.items }}
        {{- if and (eq "api-gateway-nginx" $srv.name ) (hasPrefix "keptn-" ( get $srv.metadata.labels "app.kubernetes.io/part-of" )) }}
            {{- $detectedKeptnApiGateways = append $detectedKeptnApiGateways $srv }}
        {{- end }}
    {{- end }}

    {{- if eq (len $detectedKeptnApiGateways) 0 }}
        {{- fail "Unable to detect Kepn in the kubernetes cluster!" }}
    {{- end }}
    {{- if gt (len $detectedKeptnApiGateways) 1 }}
        {{- fail (printf "Detected more than one Keptn installation: %+v" $detectedKeptnApiGateways) }}
    {{- end }}

    {{- (index $detectedKeptnApiGateways 0).metadata.namespace }}
```